### ISSUE TITLE
(Bug report) consistency check breaks search for protected notes

### ISSUE THREAD

**Issue by Nriver** [zadam/trilium#3672]
### Trilium Version

0.59.1

### What operating system are you using?

Other Linux

### What is your setup?

Local + server sync

### Operating System Version

Manjaro

### Description

My protected notes can note be searched anymore. It just throw an error.

```
ERROR: get /api/quick-search/:searchString threw exception: 'Cannot read properties of null (reading 'toString')', stack: TypeError: Cannot read properties of null (reading 'toString')
   at NoteContentFulltextExp.preprocessContent (/opt/trilium-cn/resources/app.asar/src/services/search/expressions/note_content_fulltext.js:105:43)
   at NoteContentFulltextExp.findInText (/opt/trilium-cn/resources/app.asar/src/services/search/expressions/note_content_fulltext.js:72:24)
   at NoteContentFulltextExp.execute (/opt/trilium-cn/resources/app.asar/src/services/search/expressions/note_content_fulltext.js:48:18)
   at OrExp.execute (/opt/trilium-cn/resources/app.asar/src/services/search/expressions/or.js:32:49)
   at AndExp.execute (/opt/trilium-cn/resources/app.asar/src/services/search/expressions/and.js:26:42)
   at findResultsWithExpression (/opt/trilium-cn/resources/app.asar/src/services/search/services/search.js:155:32)
   at Object.findResultsWithQuery (/opt/trilium-cn/resources/app.asar/src/services/search/services/search.js:254:12)
   at quickSearch (/opt/trilium-cn/resources/app.asar/src/routes/api/search.js:61:41)
   at Function.cb (/opt/trilium-cn/resources/app.asar/src/routes/routes.js:409:34)
   at Function.sqliteTransaction (/opt/trilium-cn/resources/app.asar/node_modules/better-sqlite3/lib/methods/transaction.js:65:24)

JS Error: 500 GET /api/quick-search/test - Cannot read properties of null (reading 'toString')
Stack: Error
   at Object.N [as logError] (http://127.0.0.1:37840/assets/v0.59.1/app-dist/desktop.js:2:101211)
   at Object.throwError (http://127.0.0.1:37840/assets/v0.59.1/app-dist/desktop.js:2:86320)
   at h (http://127.0.0.1:37840/assets/v0.59.1/app-dist/desktop.js:2:82301)
   at async EventEmitter.<anonymous> (http://127.0.0.1:37840/assets/v0.59.1/app-dist/desktop.js:2:83170)
```

Here is the code goes wrong. It invokes a `toString()` function.
![code0](https://user-images.githubusercontent.com/6752679/223005636-f3245a48-2547-4ca4-a244-3feb8282b1b3.png)

After some debug, I get the bad note ID. Searched the data in the database.
![db](https://user-images.githubusercontent.com/6752679/223005223-3eb92b4e-68f3-4ba3-bab1-4b7f63cc4edf.png)

The date leads me to this code for consistency check.
![code1](https://user-images.githubusercontent.com/6752679/223005385-e8b87e7a-74c5-4e1d-a5a9-123d401d5d90.png)

The default value for the protected notes is null.
![code2](https://user-images.githubusercontent.com/6752679/223006134-a48fd3a8-7840-4a2d-83c9-7f6ea4b6b776.png)

Appearently, you can not invoke `toString()` for that.

**Comment by zadam** [comment]
Hello, could you check your logs for "content was set to empty string since there was no corresponding row" to see if this really happened to this note?

**Comment by Nriver** [comment]
I'll check it later as I use trilium on several different computers, couldn't recall which one caused this.

**Comment by Nriver** [comment]
Found it in the log several days ago. The note id matches.

![log](https://user-images.githubusercontent.com/6752679/223412772-953adfce-1562-4198-a3ad-f474e91a2513.png)

I think I was putting a dozen of images into Trilium and happens to make two client sync at the same time when the network was not very good. I cancelled the sync before it finished, and later there were some images get broken. Then I turned to consistent check.
